cli/src/command.rs

```diff
@@ -215,6 +215,68 @@ fn ensure_dev(spec: &Box<dyn service::ChainSpec>) -> std::result::Result<(), Str
 	}
 }
 
+/// Runs a performance check via compiling sample wasm code with a timeout.
+/// Only available in release build since the check would take too much time otherwise.
+/// Returns `Ok` if the check has been passed previously.
+#[cfg(not(debug_assertions))]
+fn host_perf_check() -> Result<()> {
+	use polkadot_node_core_pvf::sp_maybe_compressed_blob;
+	use std::{fs::OpenOptions, path::Path, time::Duration};
+
+	const PERF_CHECK_TIME_LIMIT: Duration = Duration::from_secs(20);
```

Comment:
This should be derived as multiplicator with an expected bench duration based on our minimum viable host.

Reply:
The value in PVF host was increased recently to 60 seconds. The question is: do we want to give big enough time window for every machine or actually filter out some of them in this "benchmark"? 10 seconds seemed OK but it turned out that as a host gets overwhelmed with prepare requests this time goes up. Don't we want to avoid it?

I think @pepyakin could suggest a proper value here.

Reply:
My point being, this cannot be hardcoded, but should always depend upon the reference value we generate separately on our bench machine, which then in turn should be included here. _How_ this is implemented and the values are obtained is up for discussion.